### PR TITLE
Fix MsgDropRate missing from NonPersistentTopics stats output.

### PR DESCRIPTION
**Pull request by MarvinCai**
Fixes #https://github.com/apache/pulsar/issues/10495

### Motivation
MsgDropRate info is missing after NonPersistentTopics admin api merged with Topics admin api. This PR is trying to fix this.

### Modifications
Seems due to API merging, data is not properly deserialized in admin client.
And also due to the added TopicsStats interface, the field hiding causing weird behavior with Jackson so fields in NonPersistentTopicStatsImpl intended to hide superclass' fields are not shown in output.

Fixing by not using same field name to hide superclass fields and use @JsonIgnore to hide them from output. And add new fields to store subscription/publisher/replicator info for NonPersistentTopic.
This does change the output name of those info, but it only changed in cli output, for admin client the old getSubscriptions/getSubscriptions/getReplication will still work. 
```
{
  "msgRateIn" : 0.0,
  "msgThroughputIn" : 0.0,
  "msgRateOut" : 0.0,
  "msgThroughputOut" : 0.0,
  "bytesInCounter" : 0,
  "msgInCounter" : 0,
  "bytesOutCounter" : 0,
  "msgOutCounter" : 0,
  "averageMsgSize" : 0.0,
  "msgChunkPublished" : false,
  "storageSize" : 0,
  "backlogSize" : 0,
  "offloadedStorageSize" : 0,
  "lastOffloadLedgerId" : 0,
  "lastOffloadSuccessTimeStamp" : 0,
  "lastOffloadFailureTimeStamp" : 0,
  "waitingPublishers" : 0,
  "nonContiguousDeletedMessagesRanges" : 0,
  "nonContiguousDeletedMessagesRangesSerializedSize" : 0,
  "msgDropRate" : 0.0,
  "publishers" : [ ],
  "subscriptions" : {
    "my-sub" : {
      "msgRateOut" : 0.0,
      "msgThroughputOut" : 0.0,
      "bytesOutCounter" : 0,
      "msgOutCounter" : 0,
      "msgRateRedeliver" : 0.0,
      "chunkedMessageRate" : 0,
      "msgBacklog" : 0,
      "backlogSize" : 0,
      "msgBacklogNoDelayed" : 0,
      "blockedSubscriptionOnUnackedMsgs" : false,
      "msgDelayed" : 0,
      "unackedMessages" : 0,
      "type" : "Exclusive",
      "msgRateExpired" : 0.0,
      "totalMsgExpired" : 0,
      "lastExpireTimestamp" : 0,
      "lastConsumedFlowTimestamp" : 0,
      "lastConsumedTimestamp" : 0,
      "lastAckedTimestamp" : 0,
      "lastMarkDeleteAdvancedTimestamp" : 0,
      "consumers" : [ {
        "msgRateOut" : 0.0,
        "msgThroughputOut" : 0.0,
        "bytesOutCounter" : 0,
        "msgOutCounter" : 0,
        "msgRateRedeliver" : 0.0,
        "chunkedMessageRate" : 0.0,
        "consumerName" : "bfddf",
        "availablePermits" : 1000,
        "unackedMessages" : 0,
        "avgMessagesPerEntry" : 1000,
        "blockedConsumerOnUnackedMsgs" : false,
        "lastAckedTimestamp" : 0,
        "lastConsumedTimestamp" : 0,
        "metadata" : { },
        "address" : "/127.0.0.1:61295",
        "connectedSince" : "2021-08-08T13:12:51.218694+08:00",
        "clientVersion" : "2.9.0-SNAPSHOT"
      } ],
      "isDurable" : false,
      "isReplicated" : false,
      "consumersAfterMarkDeletePosition" : { },
      "nonContiguousDeletedMessagesRanges" : 0,
      "nonContiguousDeletedMessagesRangesSerializedSize" : 0,
      "msgDropRate" : 0.0,
      "durable" : false,
      "replicated" : false
    }
  },
  "replication" : { }
}
```

### Verifying this change

- [ ] Make sure that the change passes the CI checks.
This change added tests and can be verified as follows:
- Added unit test.

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): no
  - The public API: yes
  - The schema: no
  - The default values of configurations: no
  - The wire protocol: no
  - The rest endpoints: no
  - The admin cli options: no
  - Anything that affects deployment: no

### CODE REVIEW

File: pulsar-client-admin-api/src/main/java/org/apache/pulsar/client/admin/Topics.java [hunk 1]
```diff
@@ -849,6 +850,62 @@ default TopicStats getStats(String topic) throws PulsarAdminException {
...
+     * @return the topic statistics
+     *
+     * @throws NotAuthorizedException
+     *             Don't have admin permission
+     * @throws NotFoundException
+     *             Topic does not exist
+     * @throws PulsarAdminException
+     *             Unexpected error
+     */
+    NonPersistentTopicStats getStatsNonPersistent(String topic, boolean getPreciseBacklog,
```
Comment (codelipenghui) [inline pulsar-client-admin-api/src/main/java/org/apache/pulsar/client/admin/Topics.java:873]: The NonPersistentTopicStats also implement the TopicStats, So I think we do not need to introduce a new method in the API? We can just use the `getStats` method, and if users want to get the msgDropRate, they can convert TopicStats to NonPersistentTopicStats directly?

Comment (MarvinCai) [reply pulsar-client-admin-api/src/main/java/org/apache/pulsar/client/admin/Topics.java:873]: @codelipenghui 
I've tried that before, once the response object got serialized as TopicStats, even we convert it to NonPersistentTopicStats later the msgDropRate field will be missing, I think it's just lost during deserialization, so seems we'll have to deserialize to the desired type or interface when we get the response(NonPersistentTopicStats).

Comment (MarvinCai) [reply pulsar-client-admin-api/src/main/java/org/apache/pulsar/client/admin/Topics.java:873]: @codelipenghui 
I just tried that, once the response object got deserialized as TopicStats, it's actual type will be TopicStatsImpl at client side(we set the binding at ObjectMapperFactory [ref](https://github.com/apache/pulsar/blob/3cbfd7454ce824a720d1dc645ea23ee3831ffff0/pulsar-common/src/main/java/org/apache/pulsar/common/util/ObjectMapperFactory.java#L179)), even server send it out as NonPersistentTopicStatsImpl, and we won't be able to convert the object back to NonPersistentTopicStats or NonPersistentTopicStatsImpl. 
I think we'll have to deserialize the response to the desired type when we receive it. So a new method is needed for this purpose.

**Comment by MarvinCai** [comment]
@codelipenghui addressed your comments, PTAL

**Comment by sijie** [comment]
@codelipenghui Can you review this?

File: pulsar-common/src/main/java/org/apache/pulsar/common/policies/data/stats/NonPersistentTopicStatsImpl.java [hunk 2]
```diff
@@ -47,31 +45,49 @@
...
+    }
+
+    /** List of connected publishers on this non-persistent topic w/ their stats. */
+    public List<? extends NonPersistentPublisherStats> nonPersistentPublishers;
+
+    /** Map of non-persistent subscriptions with their individual statistics. */
+    public Map<String, ? extends NonPersistentSubscriptionStats> nonPersistentSubscriptions;
+
+    /** Map of non-persistent replication statistics by remote cluster context. */
+    public Map<String, ? extends NonPersistentReplicatorStats> nonPersistentReplicators;
```
Comment (codelipenghui) [inline pulsar-common/src/main/java/org/apache/pulsar/common/policies/data/stats/NonPersistentTopicStatsImpl.java:76]: Can we add @JsonAlias to make sure the name that exposed to the JSON result not changed?

Comment (MarvinCai) [reply pulsar-common/src/main/java/org/apache/pulsar/common/policies/data/stats/NonPersistentTopicStatsImpl.java:80]: @JsonAlias seems only used during deserialization, used @JasonProperty to keep the output field name. Updated issue description.

**Review by codelipenghui — APPROVED** [review]

**Review by eolivelli — APPROVED** [review]

**Pull request merged**